### PR TITLE
ticket 10 completed

### PR DESCRIPTION
**Pull request by chloewong108**

### CODE REVIEW

File: controllers/news.controllers.js [hunk 1]
```diff
@@ -67,3 +68,18 @@ exports.getAllCommentsById = (req, res, next) => {
       next(err);
     });
 };
+
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+  selectArticleId(article_id)
+    .then((response) => {
+      return addComment(article_id, body, username);
```
Comment (pkaodev) [inline controllers/news.controllers.js:77]: nice work with reusing function here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -232,3 +232,60 @@ describe("/api/articles/:article_id/comments", () => {
...
+          expect(body.comment).toEqual({
+            body: "coolio",
+            votes: 0,
+            author: "butter_bridge",
+            article_id: 2,
+            comment_id: expect.any(Number),
+            created_at: expect.any(String),
+          });
+        });
+    });
```
Comment (pkaodev) [inline __tests__/app.test.js]: so *technically* you can't be sure that the comment has correctly been saved in the db with this (maybe the model does something weird, but still returns a 201).

we can test for this fully by doing a db.query in another .then block in the test which looks for this new made comment

File: __tests__/app.test.js [hunk 3]
```diff
@@ -232,3 +232,60 @@ describe("/api/articles/:article_id/comments", () => {
...
+      return request(app)
+        .post("/api/articles/1000/comments")
+        .send({ username: "butter_bridge", body: "coolio" })
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("error 404: does not exist.");
+        });
+    });
+  });
+});
```
Comment (pkaodev) [inline __tests__/app.test.js:312]: What if someone who isn't a user tries to post a comment?  let's make sure that can't happen.  you will need a test for this too.

**Review by pkaodev — CHANGES_REQUESTED** [review]
looks good, just another scenario to take into consideration (see comments)

**Comment by douglashellowell** [comment]
great!

ready to merge 💪

**Pull request merged**